sdks/python/apache_beam/io/avroio_test.py

```diff
@@ -472,10 +416,152 @@ def test_sink_transform_snappy(self):
+          len_records = len(self.RECORDS)
+          for i in range(count):
+            writer.append(self.RECORDS[i % len_records])
+          writer.close()
+          self._temp_files.append(f.name)
+          return f.name
+
+    def test_split_points(self):
+        num_records = 12000
+        file_name = self._write_data(count=num_records)
+        source = _create_avro_source(file_name, use_fastavro=self.use_fastavro)
+
+        splits = [
+            split
+            for split in source.split(desired_bundle_size=float('inf'))
+        ]
+        assert len(splits) == 1
+        range_tracker = splits[0].source.get_range_tracker(splits[0].start_position, splits[0].stop_position)
+
+        split_points_report = []
+
+        for _ in splits[0].source.read(range_tracker):
+          split_points_report.append(range_tracker.split_points())
+        # There will be a total of num_blocks in the generated test file,
+        # proportional to number of records in the file divided by syncronization
+        # interval used by avro during write. Each block has more than 10 records.
+        num_blocks = int(math.ceil(14.5 * num_records /
+                                   avro.datafile.SYNC_INTERVAL))
+        assert num_blocks > 1
+        # When reading records of the first block, range_tracker.split_points()
+        # should return (0, iobase.RangeTracker.SPLIT_POINTS_UNKNOWN)
+        self.assertEqual(
+            split_points_report[:10],
+            [(0, iobase.RangeTracker.SPLIT_POINTS_UNKNOWN)] * 10)
+
+        # When reading records of last block, range_tracker.split_points() should
+        # return (num_blocks - 1, 1)
+        self.assertEqual(split_points_report[-10:], [(num_blocks - 1, 1)] * 10)
+
+    def test_dynamic_work_rebalancing_exhaustive(self):
+        # Adjusting block size so that we can perform a exhaustive dynamic
+        # work rebalancing test that completes within an acceptable amount of time.
+        old_sync_interval = avro.datafile.SYNC_INTERVAL
+        try:
+            avro.datafile.SYNC_INTERVAL = 2
+            file_name = self._write_data(count=5)
+            source = _create_avro_source(file_name, use_fastavro=self.use_fastavro)
+            splits = [split
+                    for split in source.split(desired_bundle_size=float('inf'))]
+            assert len(splits) == 1
+            source_test_utils.assert_split_at_fraction_exhaustive(splits[0].source)
+        finally:
+            avro.datafile.SYNC_INTERVAL = old_sync_interval
+
+class TestFastAvro(AvroBase,unittest.TestCase):
+    def __init__(self, methodName='runTest'):
+        super(TestFastAvro, self).__init__(methodName)
+        self.use_fastavro = True
+        self.SCHEMA = parse_schema(ast.literal_eval('''{"namespace": "example.avro","type": "record","name": "User","fields": [{"name": "name", "type": "string"},{"name": "favorite_number",  "type": "int"}, {"name": "favorite_color", "type": "string"}]}'''))
```

Comment:
Let's add the string version of the schema to the base class, and load it here with `json.loads` instead of `ast.literal_eval`.